st2client/tests/unit/test_auth.py

```diff
@@ -41,277 +41,393 @@
 }
 
 
-class TestWhoami(base.BaseCLITestCase):
+class CaptureStdout(list):
+    """This is a bit of borrowed code to make it easier to read from stdout
+
+    It's meant to be used within a context manager.
+    """
+
+    def __enter__(self):
+        self._stdout = sys.stdout
+        sys.stdout = self._stringio = StringIO()
+        return self
+
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio    # free up some memory
+        sys.stdout = self._stdout
+
+
+class TestLoginBase(base.BaseCLITestCase):
+    '''A base class for testing related to 'st2 login' commands
+
+    This exists primarily to ensure that each specific test case is kept atomic,
+    since the tests create actual files on the filesystem - as well as to cut down
+    on duplicate code in each test class
+    '''
+
+    DOTST2_PATH = os.path.expanduser('~/.st2/')
 
     def __init__(self, *args, **kwargs):
```

Comment:
Instead of using kwargs, it would be a bit more readable to just directly declare arguments in the constructor signature.